lib/compilers/sway.ts

```diff
@@ -0,0 +1,245 @@
+                        const match = line.match(/^\s*(\d+)\s+(\d+)\s+/);
+                        if (match && symbols) {
+                            const opcodeIndex = match[1];
+                            const symbolInfo = symbols.map[opcodeIndex];
+                            if (symbolInfo && symbolInfo.path === 1) {
+                                return {
+                                    text: line,
+                                    source: {
+                                        file: symbols.paths[symbolInfo.path],
+                                        line: symbolInfo.range.start.line,
+                                        column: symbolInfo.range.start.col,
+                                        mainsource: true,
+                                    },
+                                };
+                            }
+                        }
+                        return {text: line};
+                    });
+                asm.push(...contentLines);
+            }
+        }
+
+        // Run `forc build --ir final` to gather IR output and store it in `result.irOutput`.
+        let irLines: ResultLine[] = [];
+        if (buildResult.code === 0) {
+            const irResult = await this.exec(compiler, ['build', '--ir', 'final'], {
+                ...execOptions,
+                customCwd: projectDir,
+            });
+            const lastIrMarkerIndex = irResult.stdout.lastIndexOf('// IR: Final');
+            if (lastIrMarkerIndex >= 0) {
+                const relevantIr =
+                    irResult.stdout
+                        .slice(lastIrMarkerIndex)
+                        .split('\n')
+                        .slice(1)
+                        .join('\n')
+                        .match(/(script|library|contract|predicate)\s*{[^]*?^}/m)?.[0] || '';
+                irLines = relevantIr.split('\n').map(line => ({text: line}));
+            }
+        }
+
+        // Construct and return a CompilationResult
+        const result: CompilationResult = {
+            code: buildResult.code,
+            timedOut: buildResult.timedOut ?? false,
+            stdout: splitLines(buildResult.stdout).map(line => ({text: line})),
+            stderr: splitLines(buildResult.stderr).map(line => ({text: line})),
+            asm,
+            inputFilename,
+            execTime: buildResult.execTime,
+            okToCache: true,
+            dirPath: projectDir,
+            irOutput:
+                irLines.length > 0
+                    ? {
+                          asm: irLines.map(line => ({
+                              text: line.text,
+                          })),
+                      }
+                    : undefined,
+        };
+
+        return result;
+    }
+}
+
+const FORC_TOML_CONTENT = `[project]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "godbolt"
+
+[dependencies]
+std = { git = "https://github.com/FuelLabs/sway", tag = "v0.66.7" }
```

Comment:
Is it safe to hardcode v0.66.7 here? It seems like this should be configurable for each compiler?

Reply:
good catch we can remove this line as the correct version of `std` will be pulled in by `forc` when defining the compiler version. Will remove.